tests/Gating.UnitTests/GateContextUnitTests.cs

```diff
@@ -21,6 +21,11 @@ namespace Microsoft.Omex.Gating.UnitTests
 	/// </summary>
 	public sealed class GateContextUnitTests : UnitTestBase
 	{
+#pragma warning disable SM02239 // hard coded e-mail for unit tests
```

Comment:
FYI, when I spoke to the Semmle team before, their advice was to create another build configuration that only builds product code (and not test code) and to only run this through Semmle in order to avoid false positives within unit tests.

The disadvantage of course is that as new projects are added, people can forget to add them to the new configuration and we lose Semmle coverage.

I'm not sure it's really worth doing here as you don't have many suppressions. It was different for another project we had where we had a lot of suppressions akin to this and having to always suppress them was annoying.

Reply:
Thank you for explaining, I was wondering what would be Semmle team suggestions since they wanted to avoid suppressions. As you point out it this case we don't have a lot to suppress so I'll probably go with it to avoid supporting another build configuration